bin/zopen-clean

```diff
@@ -51,60 +51,35 @@ HELPDOC
 cleanUnused()
 {
   needle=$1
-  found=$(zosfind "${ZOPEN_PKGINSTALL}" -name "*/${needle}-*" -prune -type d)
-  if [ -z "${found}" ]; then
-    printInfo "No versions of package '${needle}' found for removal"
-    return
-  fi
+  [ -z "${needle}" ] && return 
+  current=$(getCurrentVersionDir "${needle}")
 
-  if isPackageActive "${needle}"; then
-    printVerbose "Getting the current versions directory [if any]"
-    deref=$(cd "${ZOPEN_PKGINSTALL}/${needle}/${needle}" && pwd -P)
+  if [ -n "${current}" ]; then
+    current=$(basename "${current}")
   else
-    printInfo "No currently installed version of package '${needle}'. Removing all versions"
+    printInfo "No currently active version of '${needle}'; removing all versions"
   fi
-
-  i=0
-  current=0
-  # the below would be simpler, but creates a subshell so can't get the number of entries outside!
-  #  echo "${found}" | xargs | tr ' ' '\n' | while read repo; do
-  TMP_FIFO_PIPE="${HOME}/altselect.pipe"
-  [ ! -p "${TMP_FIFO_PIPE}" ] || rm -f "${TMP_FIFO_PIPE}"
-  mkfifo "${TMP_FIFO_PIPE}"
-  echo "${found}" | xargs | tr ' ' '\n' >> "${TMP_FIFO_PIPE}" &
+  cd "${ZOPEN_PKGINSTALL}/${needle}" && zosfind . -name "./*" -prune -type d |
   while read repo; do
-    printVerbose "Parsing repo: '${repo}' as '${repo#"${ZOPEN_PKGINSTALL}"/}'"
-    if [ "${repo}" = "${repo#"${ZOPEN_PKGINSTALL}"/}" ]; then
-      printVerbose "Working around possible bug in FIFO that converts initial char to alert/bell 0x07/'\a'"
-      if ! $(type od); then
-        printVerbose "Displaying erroneous string"
-        out=$(echo "${repo}" | od -cx)
-        printVerbose "String details:\n '${out}'"
-      fi
-      repo="/$(echo ${repo} | cut -b 2-)"
-      printVerbose "Repo:='${repo}'"
-      syslog "${ZOPEN_LOG_PATH}/audit.log" "${LOG_W}" "${CAT_FILE},${CAT_SYS}" "CLEAN" "cleanUnused" "FIFO character issue while reading from ${TMP_FIFO_PIPE}. Workaround applied"
-    fi
-    i=$(expr ${i} + 1)
-    if [ "${deref#"${ZOPEN_PKGINSTALL}"/}" = "${repo#"${ZOPEN_PKGINSTALL}"/}" ]; then
-      current=${i}
-      printInfo "${NC}${GREEN}${i}: ${repo#"${ZOPEN_PKGINSTALL}"/}  <-  current${NC}"
+    printVerbose "Parsing repo: '${repo}' as '${repo#./}'"
+    repo="${repo#./}"
+    if [ "${current}" = "${repo}" ]; then
+      printInfo "${NC}${GREEN}-> ${repo}  <- Current version${NC}"
     else
-      rm -rf "${repo}" > /dev/null 2>&1
-      printInfo "${i}: ${repo#"${ZOPEN_PKGINSTALL}"/} <- Removed"
+      echo rm -rf "${ZOPEN_PKGINSTALL}/${needle}/${repo}" #> /dev/null 2>&1
```

Comment:
Assuming this `echo` was left over from debugging?

Reply:
Yep - just pushed a new commit  [my testing git had become out-of-sync with my committed-git it seems!